lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -1267,6 +1269,11 @@ SPIRVValue *LLVMToSPIRV::transValueWithoutDecoration(Value *V,
     return BV ? mapValue(V, BV) : nullptr;
   }
 
+  if (InlineAsm *IA = dyn_cast<InlineAsm>(V))
+    if (BM->isAllowedToUseExtension(ExtensionID::SPV_INTEL_inline_assembly) &&
+        IA)
```

Comment:
```suggestion
    if (BM->isAllowedToUseExtension(ExtensionID::SPV_INTEL_inline_assembly))
```

No need to check `IA` - it is guaranteed to be non-`nullptr` by previous `if`